src/Test/Verilog/Pretty.idr

```diff
@@ -363,23 +363,22 @@ parameters {opts : LayoutOpts} (m : ModuleSig) (ms: ModuleSigsList)  (subMs : Fi
         True  => []
         False => warnings ++ [ "//   \{printConnType op on} -> \{printConnType np nn}" ]
 
-    setDilimiters : List (List String) -> List String
-    setDilimiters (x::[]::xs) = x ++ setDilimiters xs
-    setDilimiters ([]::y::xs) = y ++ setDilimiters xs
-    setDilimiters (x::y::xs)  = x ++ ["//"] ++ y ++ setDilimiters xs
-    setDilimiters (x::xs)     = x ++ setDilimiters xs
-    setDilimiters []          = []
-
-    printAllImplicitCasts : List SVType -> List String -> List SVType -> List String -> List $ List String
-    printAllImplicitCasts (p::ps) (n::ns) (p'::ps') (n'::ns') = printImplicitCast p n p' n' :: printAllImplicitCasts ps ns ps' ns'
+    printAllImplicitCasts : List SVType -> List String -> List SVType -> List String -> List String
+    printAllImplicitCasts (p::ps) (n::ns) (p'::ps') (n'::ns') = do
+      let curr = printImplicitCast p n p' n'
+      let rest = printAllImplicitCasts ps ns ps' ns'
+      case isNil curr of
```

Comment:
One `isNil` is checked with `case`, another with `if`, this all looks weird. I'd suggest something like `if isNil curr || isNil rest then curr ++ rest else curr ++ ["//"] ++ rest`